### PR TITLE
Delete Comment

### LINKED ISSUE #553: 🚀 Delete Comment
## :rocket: Feature

**As** a user
**I should be able to** delete my comment
**So that** I can remove my comment if I deem necessary

### PR DESCRIPTION
**Pull request by Tirokk**
## Pullrequest

Delete an own comment.

### Issues

- fixes #553

### Checklist

- [X] None

#### Unit Tests

```bash
# in folder webapp run
$ yarn test
```

```bash
# in folder backend run
$ yarn test:jest
```
### Todo

- [x] Unit tests
- [x] Update list
- [x] customize mutation `DeleteComment`
- [x] Backend tests for mutation `DeleteComment`
- [ ] Cypress tests

### CODE REVIEW

**Comment by Tirokk** [comment]
I have refactored files `DeleteModal.vue`, `DeleteModal.spec.js` and the localisations to make it work for Post and Comment.

**Comment by mattwr18** [comment]
hey @Tirokk, so close!! you might have seen that it's failing the build on this test
`components/Modal/DisableModal.spec.js` with this output `Missing required prop: "callbacks"`

I think you know how to get passed this, but let me know if you'd like to pair on it

File: webapp/components/PostCard/index.vue [hunk 1]
```diff
@@ -74,11 +82,16 @@ export default {
...
   },
+  mixins: [PostMutationHelpers],
   props: {
     post: {
       type: Object,
       required: true,
     },
+    width: {
+      type: Object,
+      default: () => {},
```
Comment (roschaefer) [inline webapp/components/PostCard/index.vue]: Oh, ok, so `() => {}` creates an empty object. So, I guess the syntax for empty function is `() => () => {}` :thinking:

File: webapp/mixins/PostMutationHelpers.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        this.$toast.success(this.$t('delete.contribution.success'))
+        // console.log('called "this.$t" !!!')
+        switch (postDisplayType) {
+          case 'list':
+            this.$emit('deletePost')
+            // console.log('emitted "deletePost" !!!')
+            break
+          default:
+            // case 'page'
+            // console.log('called "this.$router.history.push" !!!')
```
Comment (roschaefer) [inline webapp/mixins/PostMutationHelpers.js]: Remove commented code.

File: webapp/mixins/PostMutationHelpers.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import gql from 'graphql-tag'
+
+export default {
+  methods: {
+    async deletePostCallback(postDisplayType = 'list') {
```
Comment (roschaefer) [inline webapp/mixins/PostMutationHelpers.js]: As I said, try to avoid mixins. I guess this one might be replaced in a future refactoring. It's OK for now.

**Comment by roschaefer** [comment]
@Tirokk Oh, and one more thing: Try to avoid unrelated changes. I guess your built-in linter in VS-Code has formatted a lot of files. In general, changes that dont' change anything will make it harder for your code reviewer.

**Pull request merged**